Subject: [PATCH] llvm/cmake: fix installation destination for standalone MLIR

This isn't suitable for normal LLVM use, as that should be installed
into LLVM_TOOLS_INSTALL_DIR (although they're often the same).

However, installing to LLVM's tool directory from a standalone MLIR
build is problematic -- MLIR's bits go in its own bin directory.

We could fix this with some conditionals, but lets leave our LLVM alone
for now (and not rebuild it just for this change for MLIR).
---
diff --git a/TableGen.cmake b/TableGen.cmake
index 59fd0d3b0733..d0f0bd8da170 100644
--- a/TableGen.cmake
+++ b/TableGen.cmake
@@ -192,7 +192,7 @@ macro(add_tablegen target project)
     install(TARGETS ${target}
             ${export_to_llvmexports}
             COMPONENT ${target}
-            RUNTIME DESTINATION ${LLVM_TOOLS_INSTALL_DIR})
+            RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
     if(NOT LLVM_ENABLE_IDE)
       add_llvm_install_targets("install-${target}"
                                DEPENDS ${target}
-- 
2.35.1

